### ISSUE TITLE
[question] How did you get systemd-dbus java binding?

### ISSUE THREAD

**Issue by splatch** [thjomnx/java-systemd#5]
Hey Markus,
This issue is question, so feel free to close it if you wish. I noticed that you have a bunch of code which seems to be maintained manually. Yet looking at it amount I believe first portion could be generated few years back. Was it a case? Can you recommend any code generator which actually works?
Generator I found in dbus-java produces incomplete results.

Thanks,
Łukasz

**Comment by thjomnx** [comment]
Hi,

yes, basically all of the code you spotted is maintained manually. The reason is, that the generated code of dbus-java has never been that efficient to suite my goals. Also, such generated code is in many cases not "clean", i.e. generic variable names and getters do not tell enough to a developer.

I'm not aware of any other generator that would fit here.

When I find the time then I use to try to align these parts to the latest  version of systemd.

Cheers
  Markus

**Comment by shanemikel** [comment]
Hello,  I also encountered some issues generating Systemd bindings with `dbus-java`.
I opened an issue at hypfvieh/dbus-java#195.

Your [README.md](https://github.com/thjomnx/java-systemd/blob/6364ceea049aeeb63a1f9846a2cdc87b90ea7a6b/README.md) mentions contributions by @brett-smith to use hypfvieh/dbus-java v3.
I assume it is still the case you are not using the code generator?

I am considering using your library instead of generating bindings myself.

**Comment by thjomnx** [comment]
> I assume it is still the case you are not using the code generator?

Correct, the bindings are still coded manually (and supposed to be aligned to the evolution of systemd regularly - but I don't find the time for this atm).

> Did you ever try using the QtDBus bindings module, qtjambi-dbus, in the [QtJambi Qt bindings for Java](https://github.com/OmixVisualization/qtjambi/blob/1c9295bc44b4db23b3a94d6d0e66dd938bbe016f/www/Modules.md) project?

Nope.
